From 815fb0a017fc628fe450b37b2f98cc8b80845898 Mon Sep 17 00:00:00 2001
From: Venkatesh Shukla <venkatesh.shukla.eee11@iitbhu.ac.in>
Date: Sun, 17 Aug 2014 18:09:26 +0530
Subject: [PATCH 1/7] Build subsurface_jni for building on android

For building on android subsurface downloader, we would need only the core
libraries of subsurface. All the unnecessary stuff is hence removed for
android_jni build.

Signed-off-by: Venkatesh Shukla <venkatesh.shukla.eee11@iitbhu.ac.in>
---
 subsurface.pro | 137 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 137 insertions(+)

diff --git a/subsurface.pro b/subsurface.pro
index 0aa6a5d..5cc0dbe 100644
--- a/subsurface.pro
+++ b/subsurface.pro
@@ -325,3 +325,140 @@ include(subsurface-install.pri)
 # to build debuggable binaries on Windows, you need something like this
 #QMAKE_CFLAGS_RELEASE=$$QMAKE_CFLAGS_DEBUG -O0 -g
 #QMAKE_CXXFLAGS_RELEASE=$$QMAKE_CXXFLAGS_DEBUG -O0 -g
+
+android_jni {
+	TARGET = subsurface_jni
+	TEMPLATE = lib
+	CONFIG += staticlib
+	QT -= core gui network svg concurrent
+	lessThan(QT_MAJOR_VERSION, 5) {
+		QT -= webkit
+	} else {
+		QT -= androidextras
+	}
+	INCLUDEPATH -= qt-ui $$PWD
+	DEPENDPATH -= qt-ui
+
+	HEADERS -= \
+		save-html.h \
+		worldmap-save.h \
+		worldmap-options.h \
+		qt-ui/about.h \
+		qt-ui/completionmodels.h \
+		qt-ui/divecomputermanagementdialog.h \
+		qt-ui/divelistview.h \
+		qt-ui/divepicturewidget.h \
+		qt-ui/diveplanner.h \
+		qt-ui/downloadfromdivecomputer.h \
+		qt-ui/graphicsview-common.h \
+		qt-ui/globe.h \
+		qt-ui/kmessagewidget.h \
+		qt-ui/maintab.h \
+		qt-ui/mainwindow.h \
+		qt-ui/modeldelegates.h \
+		qt-ui/models.h \
+		qt-ui/preferences.h \
+		qt-ui/printdialog.h \
+		qt-ui/printlayout.h \
+		qt-ui/printoptions.h \
+		qt-ui/simplewidgets.h \
+		qt-ui/starwidget.h \
+		qt-ui/subsurfacewebservices.h \
+		qt-ui/tableview.h \
+		qt-ui/divelogimportdialog.h \
+		qt-ui/tagwidget.h \
+		qt-ui/groupedlineedit.h \
+		qt-ui/usermanual.h \
+		qt-ui/profile/profilewidget2.h \
+		qt-ui/profile/diverectitem.h \
+		qt-ui/profile/divepixmapitem.h \
+		qt-ui/profile/divelineitem.h \
+		qt-ui/profile/divetextitem.h \
+		qt-ui/profile/animationfunctions.h \
+		qt-ui/profile/divecartesianaxis.h \
+		qt-ui/profile/diveplotdatamodel.h \
+		qt-ui/profile/diveprofileitem.h \
+		qt-ui/profile/diveeventitem.h \
+		qt-ui/profile/divetooltipitem.h \
+		qt-ui/profile/ruleritem.h \
+		qt-ui/updatemanager.h \
+		qt-ui/divelogexportdialog.h \
+		qt-ui/usersurvey.h
+
+	SOURCES += linux.c
+
+	SOURCES -= \
+		gettextfromc.cpp \
+		main.cpp \
+		android.cpp \
+		divecomputer.cpp \
+		worldmap-save.c \
+		save-html.c \
+		qt-gui.cpp \
+		qthelper.cpp \
+		exif.cpp \
+		subsurfacesysinfo.cpp \
+		qt-ui/about.cpp \
+		qt-ui/completionmodels.cpp \
+		qt-ui/divecomputermanagementdialog.cpp \
+		qt-ui/divelistview.cpp \
+		qt-ui/divepicturewidget.cpp \
+		qt-ui/diveplanner.cpp \
+		qt-ui/downloadfromdivecomputer.cpp \
+		qt-ui/globe.cpp \
+		qt-ui/graphicsview-common.cpp \
+		qt-ui/kmessagewidget.cpp \
+		qt-ui/maintab.cpp \
+		qt-ui/mainwindow.cpp \
+		qt-ui/modeldelegates.cpp \
+		qt-ui/models.cpp \
+		qt-ui/preferences.cpp \
+		qt-ui/printdialog.cpp \
+		qt-ui/printlayout.cpp \
+		qt-ui/printoptions.cpp \
+		qt-ui/simplewidgets.cpp \
+		qt-ui/starwidget.cpp \
+		qt-ui/subsurfacewebservices.cpp \
+		qt-ui/tableview.cpp \
+		qt-ui/divelogimportdialog.cpp \
+		qt-ui/tagwidget.cpp \
+		qt-ui/groupedlineedit.cpp \
+		qt-ui/usermanual.cpp \
+		qt-ui/updatemanager.cpp \
+		qt-ui/divelogexportdialog.cpp \
+		qt-ui/usersurvey.cpp \
+		qt-ui/profile/profilewidget2.cpp \
+		qt-ui/profile/diverectitem.cpp \
+		qt-ui/profile/divepixmapitem.cpp \
+		qt-ui/profile/divelineitem.cpp \
+		qt-ui/profile/divetextitem.cpp \
+		qt-ui/profile/animationfunctions.cpp \
+		qt-ui/profile/divecartesianaxis.cpp \
+		qt-ui/profile/diveplotdatamodel.cpp \
+		qt-ui/profile/diveprofileitem.cpp \
+		qt-ui/profile/diveeventitem.cpp \
+		qt-ui/profile/divetooltipitem.cpp \
+		qt-ui/profile/ruleritem.cpp \
+		qt-ui/profile/tankitem.cpp
+
+	FORMS -= \
+		qt-ui/about.ui \
+		qt-ui/divecomputermanagementdialog.ui \
+		qt-ui/diveplanner.ui \
+		qt-ui/downloadfromdivecomputer.ui \
+		qt-ui/maintab.ui \
+		qt-ui/mainwindow.ui \
+		qt-ui/preferences.ui \
+		qt-ui/printoptions.ui \
+		qt-ui/renumber.ui \
+		qt-ui/shifttimes.ui \
+		qt-ui/shiftimagetimes.ui \
+		qt-ui/webservices.ui \
+		qt-ui/tableview.ui \
+		qt-ui/divelogimportdialog.ui \
+		qt-ui/searchbar.ui \
+		qt-ui/divelogexportdialog.ui \
+		qt-ui/plannerSettings.ui \
+		qt-ui/usersurvey.ui
+}
+
-- 
1.9.3

